adr/minimal-sdk.md

```diff
@@ -0,0 +1,53 @@
+
+## Decision
+
+We will adopt **Option 7**. We will create a **new, minimal, lightweight Red Hat Agent SDK** and concurrently establish a strategy to **officially recommend and ensure compatibility with established third-party frameworks**.
+
+The new Red Hat Agent SDK will be written in **Python**. The specific list of third-party frameworks we will recommend and ensure compatibility with will be addressed in one or more future ADRs.
+
+The key characteristics of the Red Hat Agent SDK will be:
+
+* **Enablement, Not Competition:** Its primary purpose is to serve as a clear, simple "golden path" for new users, tutorials, documentation, and basic proof-of-concept applications.
+* **Minimalist Scope:** It will focus exclusively on implementing the core agent loop (state management, API calls, tool dispatching) as a reference. It is not intended to have complex features like chains, memory modules, or vector store integrations.  The details of the scope will be defined by the capabilities we want to introduce in our enablement and/or any automatic generation of agentic code that we do.
+* **Reference Implementation:** It will act as the reference for how to correctly interact with the Llama Stack Responses API in an agentic fashion, providing a foundation that other tools can build upon or refer to.  It may be useful as a reference for developers who want to build their own Agent abstraction for a single application.
+
+This two-pronged approach (a minimal "first-step" SDK and a clear "next-step" recommendation of powerful external tools) provides the best balance of speed, user choice, and focused engineering effort.
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+* **Rapid Time-to-Market:** A minimal SDK is a fast way to deliver a functional agent construct to our users, allowing us to provide a complete end-to-end story quickly.
+* **Empowers User Choice:** This strategy explicitly embraces the rich AI ecosystem. Customers with existing expertise in established frameworks can use their preferred tools, while new users have a simple, non-intimidating entry point.
+* **Low Engineering Overhead:** By strictly limiting the scope of our own SDK, we avoid the massive maintenance burden of creating and supporting a full-featured framework. Engineering efforts are focused and contained.
+* **Simplified Tooling:** Having a single, simple, Red Hat-owned SDK as a target greatly simplifies the development of any integrated tooling (e.g., agent creation wizards in the RHOAI console).
+* **Avoids Containerization Complexity:** Choosing an SDK over a container image avoids the overhead of building, distributing, securing, and running a container for what is fundamentally client-side logic, simplifying the developer workflow.
```

Comment:
I don't understand this. An agent needs to be deployed in order to run. Is the argument that by focusing on an SDK we push the burden of building, securing and running a container for that agent onto users? Does that make sense when we are primarily a platform for running AI workloads?

Reply:
The servers will of course be containerized in RHOAI (but not RHEL AI).  This is about whether we need the client-side logic to also be containerized.  It is certainly *possible* to containerize the client side logic too, but this logic is a thin layer around calls to the server APIs.  So putting that into a container too seems more complex than we really need.

Reply:
My point is, if it is to be deployed on RHOAI it will *need* to be containerised. So how is this "simplifying the developer workflow"?

Reply:
My assumption for a container based approach is that you would not deploy both an agent and an application. Either you would base the application container on the agent and thus combine them, or you would invoke the agent from some other deployed application.

However I feel this has become a tangential distraction that isn't really relevant to the main decision, so I withdraw my comment.